src/Symfony/Component/Lock/Store/SemaphoreStore.php

```diff
@@ -75,16 +75,26 @@ private function lock(Key $key, $blocking)
             return;
         }
 
-        $resource = sem_get(crc32($key));
+        $keyId = crc32($key);
+        $resource = sem_get($keyId);
 
         if (\PHP_VERSION_ID < 50601) {
             if (!$blocking) {
                 throw new NotSupportedException(sprintf('The store "%s" does not supports non blocking locks.', get_class($this)));
             }
 
-            $acquired = sem_acquire($resource);
+            $acquired = @sem_acquire($resource);
+            while (!$acquired) {
+                $resource = sem_get($keyId);
+                $acquired = @sem_acquire($resource);
+            }
         } else {
-            $acquired = sem_acquire($resource, !$blocking);
+            $acquired = @sem_acquire($resource, !$blocking);
+
+            while ($blocking && !$acquired) {
+                $resource = sem_get($keyId);
+                $acquired = @sem_acquire($resource, !$blocking);
+            }
         }
 
         if (!$acquired) {
```

Comment:
Just wondering: can this happen at all now? do we need some sort of timeout in the while above?

Reply:
yes, if `\PHP_VERSION_ID >= 50601` and `!$blocking`